app/routes/goal_routes.py

```diff
@@ -1 +1,118 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from ..db import db
+from ..models.goal import Goal
+from ..models.task import Task
+from .route_utilities import validate_model
+from datetime import datetime, timezone
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+
+@goals_bp.post("")
+def create_goal():
+
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal.from_dict(request_body)
+    except KeyError as e:
+        if e.args[0] == "title":
+            response = {"details": "Invalid data"}
+        return make_response(response, 400)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response = {"goal": new_goal.create_response()}
+
+    return response, 201
+
+
+@goals_bp.post("/<goal_id>/tasks")
+def create_tasks_with_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    task_ids = request_body["task_ids"]
+    for task_id in task_ids:
+        task = validate_model(Task, task_id)
+        task.goal_id = goal.id
+
+        db.session.commit()
+
+    response = {"id": goal.id}
+    response.update(request_body)
+    return make_response(response)
+
+
+@goals_bp.get("")
+def get_all_goals():
+    query = db.select(Goal)
+    title_param = request.args.get("title")
+
+    query = query.order_by(Goal.id)
+    goals = db.session.scalars(query).all()
+    response = [goal.create_response() for goal in goals]
+
+    return response
+
+
+@goals_bp.get("/<goal_id>")
+def get_one_goal(goal_id):
+    try:
+        goal = validate_model(Goal, goal_id)
+    except:
+        return make_response({"details": f"Goal {goal_id} not found"}, 404)
+
+    response = {"goal": goal.create_response()}
+
+    return response
```

Comment:
Since your implementation of `vaidate_model` only raises a generic exception, you'll need to use `abort here` to raise an HTTP Exception. `make_response` returns a Response Object to the client but we want it to be known that an exception occurred that's why we used `abort` in our Flasky example.

I'd prefer all the error handling to be done in validate_model so that this route can be concise and have just one responsibility. 

Have a look at how `validate_model` can handle the scenario where a request sends along an invalid id or an id not in the db and how it aborts an HTTP request, see [here](https://github.com/Ada-C22/flasky/blob/main/app/routes/route_utilities.py#L4)

```suggestion
def get_one_goal(goal_id):
        goal = validate_model(Goal, goal_id)
        return {"goal": goal.create_response()}
```